src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.cs

```diff
@@ -184,5 +190,22 @@ protected override void GetUnusedNamespaceImports(SemanticModel model, HashSet<S
                 }
             }
         }
+
+        private static bool IsTupleInDeconstruction(SyntaxNode tuple)
+        {
+            Contract.Assert(tuple.IsKind(SyntaxKind.TupleExpression));
+            var parent = tuple.Parent;
+            do
+            {
+                if (parent.IsKind(SyntaxKind.SimpleAssignmentExpression))
+                {
+                    return true;
+                }
+            }
+            while ((tuple = parent.Parent).IsKind(SyntaxKind.TupleExpression)
+                && (parent = parent.Parent.Parent) != null);
```

Comment:
indeed...

```if (parent.IsKind(SyntaxKind.SimpleAssignmentExpression))```

Do you have to validate which side of the assignment you're on?  Or is it intentional that your'e checking for a tuple on either side of a assignment?  If so, can you document that.

I also now hate the binary test with two assignments in it.  Just pull the assignments into the body, and do the binary test in teh while.